apps/aehttpserver/web/controllers/tx_controller.ex

```diff
@@ -0,0 +1,104 @@
+defmodule Aehttpserver.TxController do
+  use Aehttpserver.Web, :controller
+  alias Aecore.Txs.Pool.Worker, as: Pool
+  alias Aecore.Utils.Serialization, as: Serialization
+  alias Aecore.Structures.TxData
+
+  def show(conn, params) do
+    account_bin =
+      params["account"]
+      |> Base.decode16!()
+    user_txs = Pool.get_txs_for_address(account_bin)
+    case user_txs do
+      [] -> json(conn, [])
+      _ ->
+        case params["include_proof"]  do
+          "true" ->
+            blocks_for_user_txs =
+            for user_tx <- user_txs do
+              Pool.get_block_by_txs_hash(user_tx.txs_hash)
+            end
+            merkle_trees =
+            for block <- blocks_for_user_txs do
+              build_tx_tree(block.txs)
+            end
+            user_txs_trees = Enum.zip(user_txs, merkle_trees)
+            proof =
+            for user_tx_tree <- user_txs_trees  do
+              {tx, tree} = user_tx_tree
+              transaction = :erlang.term_to_binary(
+                tx
+                |> Map.delete(:txs_hash)
+                |> Map.delete(:block_hash)
+                |> Map.delete(:block_height)
+                |> Map.delete(:signature)
```

Comment:
Please refactor the code so we separate the transaction data used in hashing from other data. 

The hashing should be moved to a function in aeutil or some better place.